File: python/bob/bob.py
from string import ascii_uppercase, whitespace, ascii_uppercase


def response(hey_bob: str):
    is_all_caps = hey_bob == hey_bob.upper() and any([c for c in hey_bob if c in ascii_uppercase])
    if not hey_bob or hey_bob.split() == []:
        return 'Fine. Be that way!'
    if hey_bob[-1] == '?' and is_all_caps:
        return "Calm down, I know what I'm doing!"
    if hey_bob.strip()[-1] == '?':
        return 'Sure.'
    if is_all_caps:
        return 'Whoa, chill out!'
    return 'Whatever.'

    
if __name__ == '__main__':
    print(response('WHAT\'S GOING ON?'))

File: javascript/bob/bob.js
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const hey = (message) => {
    const msg = message.trim();
    if (msg.length == 0) {
        return 'Fine. Be that way!';
    }
    const is_yelling = (msg.toUpperCase() == msg) && (/[A-Z]/g).test(msg);
    const is_question = msg.slice(-1) == '?';
    if (is_yelling && is_question) {
        return "Calm down, I know what I'm doing!";
    }
    if (is_yelling) {
        return 'Whoa, chill out!';
    }
    if (is_question) {
        return 'Sure.';
    }
    return 'Whatever.';
};
